test/cli/test-other.py

```diff
@@ -859,3 +860,47 @@ def test_build_dir_j_memleak(tmpdir): #12111
     ]
 
     assert_cppcheck(args, ec_exp=0, err_exp=[], out_exp=out_lines)
+
+
+def test_premium_with_relative_path(tmpdir):
+    product_name = 'Cppcheck Premium ' + str(time.time())
+
+    test_file = os.path.join(tmpdir, 'test.cpp')
+    with open(test_file, 'wt') as f:
+        f.write('int main() {}')
+
+    cppcheck_exe = copy_and_prepare_cppcheck(tmpdir)
+
+    # we should not recognize custom args without cfg
+    exitcode, stdout, stderr = cppcheck([cppcheck_exe, '--premium=misra-c++-2008', test_file])
+    assert stderr == ''
+    assert stdout == 'cppcheck: error: unrecognized command line option: "--premium=misra-c++-2008".\n'
+    assert exitcode == 1
+
+    # adding cfg
+    test_cfg = tmpdir.join('cppcheck.cfg')
+    with open(test_cfg, 'wt') as f:
+        f.write("""
+                {
+                    "addons": [],
+                    "productName": "NAME",
+                      "about": "Cppcheck Premium 1.2.3.4"
+                }
+                """.replace('NAME', product_name))
+
+    # should be fine now
+    exitcode, stdout, stderr = cppcheck([cppcheck_exe, '--premium=misra-c++-2008', test_file])
+    assert stderr == ''
+    out_lines = [
+        'Checking {} ...'.format(test_file)
+    ]
+    lines = stdout.splitlines()
+    assert lines == out_lines
+    assert exitcode == 0
+
+    # check the version it should be as in cfg
+    exitcode, stdout, stderr = cppcheck([cppcheck_exe, '--version'])
+    assert stdout == product_name + '\n'
```

Comment:
maybe `stdout.strip() == product_name` would be somewhat better. So it doesn't care if there is "\r\n" in windows. or extra newlines.